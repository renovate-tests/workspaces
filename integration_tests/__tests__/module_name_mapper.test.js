/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @flow
 */
'use strict';

const runJest = require('../runJest');
const {extractSummary} = require('../utils');
const skipOnWindows = require('../../scripts/skip_on_windows');

// Works on windows, we just need to adjust snapshot test output
skipOnWindows.suite();

test('moduleNameMapper wrong configuration', () => {
  const {stderr, status} = runJest('module_name_mapper_wrong_config');
  const {rest} = extractSummary(stderr);

  expect(status).toBe(1);
  expect(rest).toMatchSnapshot();
});

test('moduleNameMapper correct configuration', () => {
  const {stderr, status} = runJest('module_name_mapper_correct_config');
  const {rest} = extractSummary(stderr);

  expect(status).toBe(0);
  expect(rest).toMatchSnapshot();
});
