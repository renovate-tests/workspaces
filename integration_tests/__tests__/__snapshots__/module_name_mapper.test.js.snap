// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moduleNameMapper correct configuration 1`] = `
" PASS  __tests__/index.js
  ✓ moduleNameMapping correct configuration

"
`;

exports[`moduleNameMapper wrong configuration 1`] = `
" FAIL  __tests__/index.js
  ● Test suite failed to run

    Configuration error:
    
    Unknown module in configuration option moduleNameMapper
    Please check:
    
    \\"moduleNameMapper\\": {
      \\"/\\\\.(css|less)$/\\": \\"no-such-module\\"
    }

"
`;
